File: pset6/readability/readability.py
from cs50 import get_string
text = get_string("Type your text here: ")
lettercount = 0
wordcount = 1
sentencecount = 0
for i in range(len(text)):
    if (text[i].isalpha() == True):
        lettercount += 1
    if (text[i] == " "):
        wordcount += 1
    if (text[i] == "!" or text[i] == "?" or text[i] == "."):
        sentencecount += 1
print("letters:", lettercount)
print("words:", wordcount)
print("sentences:", sentencecount)

CLindex = 0.0588 * (lettercount * 100 / wordcount) - 0.296 * (100 * sentencecount / wordcount) - 15.8

if (CLindex < 1):
    print("Before Grade 1")
elif (CLindex > 15):
    print("Grade 16+")
else:
    print("Grade:", round(CLindex))

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

//counters
int letters = 0;
int words = 1;
int sentences = 0;

int main(void)
{
    //get the text
    string text = get_string("Text:");

    //count letters
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        //count words
        else if (text[i] == ' ')
        {
            words++;
        }
        //count sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    //print words etc
    printf("letters: %i \ n", letters);
    printf("words: %i \ n", words);
    printf("sentences: %i \ n", sentences);

    // get the index
    float clindex = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences / (float) words) - 15.8;

    // round the number
    int grade = round(clindex);

    // print the grades
    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}
